File: training/openfaas-minikube/test-function/test-function/handler.py
import platform


class SayHello:

    def __init__(self):
        self.un = platform.uname()[1]

    def get(self, event):
        st = ''
        
        user = event.query.get('user', '')
        time = event.query.get('time', 0)

        if user != '':
            st += ', user: ' + user
        if time != 0:
            st += ', time: ' + time

        return {
            "statusCode": 201,
            "body": 'Hello from ' + self.un + st + '\n'
        }


def handle(event, context):

    print(event.method + " Received")
    hello = SayHello()
    return hello.get(event)

File: training/openfaas-minikube/test-function-go/test-function-go/handler.go
package function

import (
	"log"
	"net/http"
	"os"
	"strings"
)

// var shared between function calls
var hostName string

func init() {
	name, err := os.Hostname()
	if err != nil {
		panic(err)
	}
	hostName = name
}

// Handle a function invocation
func Handle(w http.ResponseWriter, r *http.Request) {
	log.Println("Method: " + r.Method)

	query := r.URL.Query()
	user := query.Get("user")
	time := query.Get("time")

	var sb strings.Builder
	sb.WriteString("Hello from " + hostName)

	if user != "" {
		sb.WriteString(", User: " + user)
	}
	if time != "" {
		sb.WriteString(", Time: " + time)
	}
	sb.WriteString("\n")

	w.WriteHeader(http.StatusOK)
	w.Write([]byte(sb.String()))
}
